nbgrader/apps/extensionapp.py

```diff
@@ -1,299 +1,35 @@
-
-        if sys.platform != 'win32' and (len(self.extra_args) == 0 or "assignment_list" in self.extra_args):
-            self.log.info("Activating assignment_list server extension")
-            self.enable_server_extension('nbgrader.nbextensions.assignment_list')
-
-            self.section = "tree"
-            self.toggle_nbextension("assignment_list/main")
-
-        self.log.info("Done. You may need to restart the Jupyter notebook server for changes to take effect.")
-
-
-class ExtensionDeactivateApp(DisableNBExtensionApp, NbGrader):
-
-    name = u'nbgrader-extension-deactivate'
-    description = u'Deactivate the nbgrader extension'
-
-    flags = {}
-    aliases = {}
-
-    examples = """
-
-    To deactivate all the nbgrader extensions:
-
-        nbgrader extension deactivate
-
-    To deactivate only a specific extension, you can pass the name of the
-    extension you want to deactivate as an argument, e.g.:
-
-        nbgrader extension deactivate create_assignment
-        nbgrader extension deactivate assignment_list
-
-    """
-
-    def _classes_default(self):
-        return [ExtensionDeactivateApp, DisableNBExtensionApp]
-
-    def _recursive_get(self, obj, key_list):
-        if obj is None or len(key_list) == 0:
-            return obj
-        return self._recursive_get(obj.get(key_list[0], None), key_list[1:])
-
-    def disable_server_extension(self, extension):
-        loader = JSONFileConfigLoader('jupyter_notebook_config.json', jupyter_config_dir())
-        try:
-            config = loader.load_config()
-        except ConfigFileNotFound:
-            config = Config()
-
-        if 'server_extensions' not in config.NotebookApp:
-            return
-        if extension not in config.NotebookApp.server_extensions:
-            return
-
-        config.NotebookApp.server_extensions.remove(extension)
-
-        # save the updated config
-        with io.open(os.path.join(jupyter_config_dir(), 'jupyter_notebook_config.json'), 'w+') as f:
-            f.write(six.u(json.dumps(config, indent=2)))
-
-    def start(self):
-        if len(self.extra_args) == 0 or "create_assignment" in self.extra_args:
-            self.log.info("Deactivating create_assignment nbextension")
-            self.section = "notebook"
-            self.toggle_nbextension("create_assignment/main")
-
-        if sys.platform != 'win32' and (len(self.extra_args) == 0 or "assignment_list" in self.extra_args):
-            self.log.info("Deactivating assignment_list server extension")
-            self.disable_server_extension('nbgrader.nbextensions.assignment_list')
-
-            self.log.info("Deactivating assignment_list nbextension")
-            self.section = "tree"
-            self.toggle_nbextension("assignment_list/main")
-
-        self.log.info("Done. You may need to restart the Jupyter notebook server for changes to take effect.")
+    $ jupyter serverextension enable --sys-prefix --py nbgrader
```

Comment:
The direction we are moving is to keep nbextension/jupyterlab extensions well separated from the server extensions. A couple of reasons for this:

* A single server could be serving notebooks to multiple frontends (notebook, nteract, jupyterlab)
* We are in the process of separating the notebook server into its own project and repo that won't share any code with the frontends
* We are also moving to a different approach for installing all of these things that will make it all easier for users. While we do this, we want to avoid making things more complicated with our existing stuff.

Obviously, in nbgrader, we could wrap all that and expose an endpoint that does it all automatically. For that I think the best approach is to create a conda-forge feedstock with post install scripts that do everything. This is what we are doing with the vega package and the install process is then trivial (no post install commands needed). Are you ok with that approach? Here is the vega-feedstock that has the needed scripts:

https://github.com/conda-forge/vega-feedstock